tests/test_ops.py

```diff
@@ -96,6 +101,21 @@ def test_add_variable(test_df_year, arg, df_func, fillna, ignore_units, append):
         assert_iamframe_equal(exp, obs)
 
 
+@pytest.mark.parametrize("append", (False, True))
+def test_add_scenario_non_si_unit(test_df_year, append):
+    df = test_df_year.rename(unit={"EJ/yr": "foo"})
+    exp = df_ops_variable(operator.add, "Sum", unit="foo", meta=test_df_year.meta)
+    args = ("Primary Energy", "Primary Energy|Coal", "Sum")
+
+    if append:
```

Comment:
I'm not a big fan of tests in general where there's an if inside the test that changes the test. I find that it makes it harder to read and understand. 
In this case I get the idea of wanting to avoid code duplication so perhaps this would make it more legible:
```python
if append:
    obs = df.copy()
    obs.add(*args, append=True)
    exp = df.append(exp)
else:
    obs = df.add(*args

assert_iamframe_equal(exp, obs)
```

this way the test ends with the same command in both ways and find it easier to read because my brain can go "ah ok we're simply preparing the data that we want to compare in two different ways but the comparison at the end is the same", as opposed to having to keep both branches of the if (admittedly small) in my mind.